.github/workflows/ci.yml

```
@@ -85,6 +65,27 @@ jobs:
+      - name: Run pytest
+        run: |
+          pytest --cov=. --cov-report=xml --cov-report=term-missing
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v4
+        with:
+          file: ./nexum_ai/coverage.xml
+          flags: python
+          name: nexum_ai-coverage
+          token: ${{ secrets.CODECOV_TOKEN }}
+          token: ${{ secrets.CODECOV_TOKEN }}
```

actionlint (1.7.9)

key "token" is duplicated in "with" section. previously defined at line:89,col:11. note that this key is case insensitive

(syntax-check)